projects/plugins/jetpack/modules/masterbar/admin-menu/class-admin-menu.php

```diff
@@ -113,20 +113,27 @@ public function add_my_home_menu() {
 	 * @param object $nudge The $nudge object containing the content, CTA, link and tracks.
 	 */
 	public function add_upsell_nudge( $nudge ) {
+		$dismiss_button = '';
+		if ( $nudge['dismissible'] ) {
+			$dismiss_button = '<span data-feature_class="%1$s" data-feature_id="%2$s" class="banner__dismiss"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" class="gridicon gridicons-cross dismissible-card__close-icon" height="24" width="24"><g><path d="M18.36 19.78L12 13.41l-6.36 6.37-1.42-1.42L10.59 12 4.22 5.64l1.42-1.42L12 10.59l6.36-6.36 1.41 1.41L13.41 12l6.36 6.36z"></path></g></svg></span>';
+			$dismiss_button = sprintf( $dismiss_button, wp_kses( $nudge['feature_class'], array() ), wp_kses( $nudge['id'], array() ) );
+		}
+
```

Comment:
Maybe we could add a test to ensure that a dismissable notification is being created instead of a non-dismissable one. What do you think?

On the other hand, shouldn't we be using Yoda conditions in the check?
`
if( true === $nudge['dismissible'] ) {
    // ...
`
Instead of:
`
if( $nudge['dismissible'] ) {
    // ...
`

Reply:
> Maybe we could add a test to ensure that a dismissable notification is being created instead of a non-dismissable one. 

Sorry, not 100% clear on what you mean here, if you mean unit test then there should already be one

> shouldn't we be using Yoda conditions in the check

The purpose of yoda conditions is so you don't accidentally assign a variable rather than comparing it. Risk here of that there is not, fine I think it is. Strict comparison, potentially change to I could.

Reply:
In this case, `dismissible` can only be boolean, so it's fine to just test for truthiness.

The Yoda condition is mostly useful to avoid assigning (`=`) instead of comparing (`==`) by mistake, but if there are no equal signs, then there is also no risk for that mistake. 🙂

Reply:
I just saw the test, sorry for the noise.

Reply:
no worries!